### PR TITLE
Update elementToFocusOnDismiss on unmount in focus trap zone

### PR DESCRIPTION
**Pull request by zoarif**
Issue:
The issue is in focus trap zone where previously focused element doesn't get updated after component Did Mount. This does not work with delete dialog where the previously focused element gets deleted from DOM so we need to focus on the new element that is being passed on as a prop elementToFocusOnDismiss.
Description of changes:
My change is that it updates the prop on componentWillUnmount as well so that it gets the new element in the prop and updates the local value.
Focus areas to test:
Its a pretty straight forward change . No major testing needed. 

(optional)

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/FocusTrapZone/FocusTrapZone.tsx [hunk 1]
```diff
@@ -156,4 +158,11 @@ export class FocusTrapZone extends BaseComponent<IFocusTrapZoneProps, {}> implem
       }
     }
   }
+
+  private _getElementToFocusOnDismiss(): HTMLElement {
```
Comment (antonlabunets) [inline packages/office-ui-fabric-react/src/components/FocusTrapZone/FocusTrapZone.tsx]: Should be a getter?

File: packages/office-ui-fabric-react/src/components/FocusTrapZone/FocusTrapZone.tsx [hunk 2]
```diff
@@ -41,7 +41,7 @@ export class FocusTrapZone extends BaseComponent<IFocusTrapZoneProps, {}> implem
   public componentDidMount() {
     let { elementToFocusOnDismiss, isClickableOutsideFocusTrap = false, forceFocusInsideTrap = true } = this.props;
 
-    this._previouslyFocusedElement = elementToFocusOnDismiss ? elementToFocusOnDismiss : document.activeElement as HTMLElement;
+    this._previouslyFocusedElement = this._getElementToFocusOnDismiss();
```
Comment (antonlabunets) [inline packages/office-ui-fabric-react/src/components/FocusTrapZone/FocusTrapZone.tsx]: Do we need this here anymore if we have it in unmount?

Comment (zoarif) [inline packages/office-ui-fabric-react/src/components/FocusTrapZone/FocusTrapZone.tsx]: moving this to unmount.

File: packages/office-ui-fabric-react/src/components/FocusTrapZone/FocusTrapZone.tsx (hunk 1 above)
Comment (zoarif) [inline packages/office-ui-fabric-react/src/components/FocusTrapZone/FocusTrapZone.tsx]: No need for this method anymore

**Pull request merged**